app/lib/package/api_export/exported_api.dart

```diff
@@ -175,6 +183,40 @@ final class ExportedApi {
     }));
   }
 
+  /// Search for stray files in [prefix]
+  ///
+  /// We detect stray files by looking at the the [_validatedCustomHeader].
+  /// Whenever we save a file we update the [_validatedCustomHeader] timestamp,
+  /// if it's older than [_updateValidatedAfter]. Thus, if something haven't
+  /// been updated for [_unvalidatedStrayFileAfter], then it's probably a stray
+  /// file that we don't understand.
+  ///
+  /// If there are stray files we don't really dare to delete them. They could
+  /// be introduced by a newer [runtimeVersion]. Or it could bug, but if that's
+  /// the case, what are the implications of deleting such files?
+  /// In all such cases, it's best alert and leave deletion of files as bug to
+  /// be fixed.
+  Future<void> _findStrayFiles(String prefix) async {
+    final validatedDeadline = clock.agoBy(_unvalidatedStrayFileAfter);
+    await _listBucket(prefix: prefix, delimiter: '', (item) async {
+      assert(item.isObject);
```

Comment:
Not sure about this assert, maybe throwing an error is better? Otherwise we could be missing some strange thing in prod that we have no test for...